### DISCUSSION TITLE
porting to another IDE

### DISCUSSION THREAD

**Discussion by Moineau54** [malware-dev/MDK-SE#317] [Ideas]
Hello guys,

I rencently tested the preview for Fleet, a jetbrains code editor that can do C#, and I wanted to know if it was possible to port MDK-SE to other IDEs and if possible, if it would be possible, when the IDE fully comes out, to have it on Fleet.

**Comment by malware-dev** [comment]
MDK1, not a chance.

MDK2, however, is IDE agnostic by nature. I am using Rider myself. _How_ it can be used, whether it can be automatic or has to be manual, will depend on the capabilities of the IDE.

See #308 for what is going on with MDK right now.

**Comment by malware-dev** [comment]
I forgot to mention, you _can_ in theory use MDK2 already but you'd have to set up the project manually.

1. Set up a new-style C# project (create a .NET 8 class library project, for example).
3. Open the project file:  
    change the TargetFramework from `net8.0` to `netframework48`  
    remove the `ImplicitUsings` and `Nullable` tags.  
    Add a tag `<RootNamespace>IngameScript</RootNamespace>`
    Add a tag `<GenerateAssemblyInfo>false</GenerateAssemblyInfo>`
5. Delete the pregenerated Class1.cs file
6. Add the nuget packages:  
    `Mal.Mdk2.PbAnalyzers`
    `Mal.Mdk2.References`
    `Mal.Mdk2.PbPackager`
5. Add a `Program` class deriving from `MyGridProgram`

You _should_ be able to work from that.

**Comment by malware-dev** [comment]
MDK2 templates are now available, see discussions.
